cylc/flow/graph_parser.py

```diff
@@ -229,7 +231,13 @@ def parse_graph(self, graph_string):
                     raise GraphParseError(
                         "trailing arrow: %s" % this_line)
             part_lines.append(this_line)
-            if (this_line.endswith(ARROW) or next_line.startswith(ARROW)):
+            if (any(
+                this_line.endswith(seq) or next_line.startswith(seq) for
+                seq in self.CONTINUATION_STRS
+            ) and not (any(
+                this_line.endswith(seq) or next_line.startswith(seq) for
+                seq in self.BAD_STRS
+            ))):
```

Comment:
Not your fault @wxtim as you've just extended the existing conditional ... but it would be good to catch the error here if one line ends with `seq` *and* the next line starts with `seq`.  Current this causes different (and somewhat hard to understand) downstream validation errors.

Also, this (correctly) fails validation with `error: trailing arrow`:
```
foo =>
# comment
```
 But this (incorrectly) passes, and results in a runtime error:
```
foo &
# comment
```

Reply:
I think that I have fixed both of these issues.